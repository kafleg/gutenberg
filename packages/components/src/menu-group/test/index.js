/**
 * External dependencies
 */
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import MenuGroup from '../';

describe( 'MenuGroup', () => {
	test( 'should render null when no children provided', () => {
		const wrapper = render( <MenuGroup /> );

		expect( wrapper.container.firstChild ).toBe( null );
	} );

	test( 'should render correctly', () => {
		const wrapper = render(
			<MenuGroup label="My group" instanceId="1">
				<p>My item</p>
			</MenuGroup>
		);
		expect( wrapper.container.firstChild ).toMatchSnapshot();
	} );
} );
