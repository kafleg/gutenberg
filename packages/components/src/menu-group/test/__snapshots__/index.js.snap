// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuGroup should render correctly 1`] = `
.emotion-0+.components-menu-group {
  margin-top: calc(4px * 2);
  padding-top: calc(4px * 2);
  border-top: 1px solid #1e1e1e;
}

.emotion-2 {
  padding: 0 calc(4px * 2);
  margin-top: calc(4px * 1);
  margin-bottom: calc(4px * 3);
  color: #757575;
  text-transform: uppercase;
  font-size: calc(0.92 * 13px);
  font-weight: 600;
  white-space: nowrap;
}

<div
  class="components-menu-group emotion-0 emotion-1"
>
  <div
    aria-hidden="true"
    class="components-menu-group__label emotion-2 emotion-1"
    id="components-menu-group-label-1"
  >
    My group
  </div>
  <div
    aria-labelledby="components-menu-group-label-1"
    class="emotion-4 emotion-1"
    role="group"
  >
    <p>
      My item
    </p>
  </div>
</div>
`;
